README.md

```diff
@@ -3,3 +3,411 @@
+    
+    public class Greeting {
+    
+        private String message;
+    
+        public Greeting(String message) {
+            this.message = message;
+        }
+    
+        public String getMessage() {
+            return message;
+        }
+    }
+    ```            
+2. Update the content of `GreetingController` to become:
+    ```
+    package org.acme.spring.web;
+    
+    import org.springframework.web.bind.annotation.GetMapping;
+    import org.springframework.web.bind.annotation.RequestMapping;
+    import org.springframework.web.bind.annotation.RequestParam;
+    import org.springframework.web.bind.annotation.RestController;
+    
+    
+    @RestController
+    @RequestMapping("/greeting")
+    public class GreetingController {
+    
+        @GetMapping
+        public Greeting hello(@RequestParam(defaultValue = "world")String name) {
+            return new Greeting("hello "+name);
+        }
+    }
+    ```    
+3. Open browser to http://localhost:8080/greeting?name=folks
+
+### Introduce a Service
+
+1. Create class `GreetingService` in the `org.acme.spring.web` package with the following content:
+    ```
+   package org.acme.spring.web;
+   
+   import org.springframework.stereotype.Service;
+   
+   public class GreetingService {
+   
+       public Greeting greet(String name){
+           return new Greeting(name);
+       }
+   }
+    ```            
+2. Update the content of `GreetingController` to become:
+    ```
+    package org.acme.spring.web;
+    
+    import org.springframework.beans.factory.annotation.Autowired;
+    import org.springframework.web.bind.annotation.GetMapping;
+    import org.springframework.web.bind.annotation.RequestMapping;
+    import org.springframework.web.bind.annotation.RequestParam;
+    import org.springframework.web.bind.annotation.RestController;
+    
+    
+    @RestController
+    @RequestMapping("/greeting")
+    public class GreetingController {
+    
+        @Autowired
+        private GreetingService greetingService;
+    
+        @GetMapping
+        public Greeting hello(@RequestParam(defaultValue = "world")String name) {
+            return greetingService.greeting("hello "+name);
+        }
+    }
+    ```    
+3. Open browser to http://localhost:8080/greeting?name=folks
+1. We get an error because we have not made the Service class a bean.
+1. Modify the service class to add an annotation to make it a bean, for instance `@Service`
+ ```
+   package org.acme.spring.web;
+   
+   import org.springframework.stereotype.Service;
+   
+   @Service
+   public class GreetingService {
+   
+       public Greeting greeting(String name){
+           return new Greeting(name);
+       }
+   }
+   ```
+1. Refresh browser
+
+
+### Externalize values to configuration file
+
+1. In the `GreetingService`, add a `message` field.
+1. Use Constructor injection with `@Value` annotation which is what the Spring world use to use.
+1. Change the `greet` method to use the message field.
```

Comment:
Number are wrong I think

Reply:
Same here: it's like this when you want a continue numbered list with code blocks between items. it is resolved correctly